src/borg/archive.py

```diff
@@ -1053,19 +1053,19 @@ def chunker(it, size):
         pi = ProgressIndicatorPercent(total=count, msg="Verifying data %6.2f%%", step=0.01)
         for chunk_infos in chunker(self.chunks.iteritems(), 100):
             chunk_ids = [chunk_id for chunk_id, _ in chunk_infos]
-            chunk_ids_iter = iter(chunk_ids)
             chunk_data_iter = self.repository.get_many(chunk_ids)
-            while True:
+            while chunk_ids:
                 pi.show()
+                chunk_id = chunk_ids.pop(0)
```

Comment:
Hmm, it would be better to reverse the list and pop(-1), because pop(0) has to reallocate the list. Effectively makes it another inner loop = O(n²).

```py
>>> timeit('ll = list(reversed(l))\nfor x in range(N-1): ll.pop()', setup='N = 100000; l = list(range(N))', number=10)
0.10201670200331137

>>> timeit('ll = list(l)\nfor x in range(N-1): ll.pop(0)', setup='N = 100000; l = list(range(N))', number=10)
14.560062063013902
```

(Surprisingly large difference)